sql/moz-fx-data-shared-prod/telemetry_derived/urlbar_search_sessions_v1/query.sql

```diff
@@ -0,0 +1,215 @@
+WITH events_unnested AS (
+  SELECT
+    DATE(submission_timestamp) AS submission_date,
+    sample_id,
+    client_info.client_id,
+    timestamp AS event_timestamp,
+    normalized_channel,
+    normalized_country_code,
+    name AS event_name,
+    CASE
+      WHEN name = 'engagement'
+        AND (
+          (
+            mozfun.map.get_key(extra, "engagement_type") IN (
+              "click",
+              "drop_go",
+              "enter",
+              "go_button",
+              "paste_go"
+            )
+          )
+        )
+        THEN 'engaged'
+      WHEN name = 'abandonment'
+        THEN 'abandoned'
+      WHEN name = 'engagement'
+        AND (
+          (
+            mozfun.map.get_key(extra, "engagement_type") NOT IN (
+              "click",
+              "drop_go",
+              "enter",
+              "go_button",
+              "paste_go"
+            )
+          )
+        )
+        THEN "annoyance"
+      ELSE NULL
+    END AS search_session_type,
+    SPLIT(mozfun.map.get_key(extra, "results"), ',')[OFFSET(0)] AS result_type,
+    SPLIT(mozfun.map.get_key(extra, "results"), ',') AS results,
+    mozfun.map.get_key(extra, "selected_result") AS selected_results,
+    mozfun.map.get_key(extra, "engagement_type") AS engagement_type,
+    mozfun.map.get_key(extra, "n_chars") AS number_of_chars_typed,
+    mozfun.map.get_key(extra, "n_results") AS num_total_results,
+    metrics,
+    metrics.uuid.legacy_telemetry_client_id AS legacy_telemetry_client_id,
+    ping_info.experiments
+  FROM
+    `moz-fx-data-shared-prod.firefox_desktop_stable.events_v1`,
+    UNNEST(events)
+  CROSS JOIN
+    UNNEST(ping_info.experiments) AS e
+  WHERE
+    DATE(submission_timestamp) = @submission_date
+    AND category = 'urlbar'
+    AND name IN ('engagement', 'abadonment')
+),
+events_summary AS (
+  SELECT
+    submission_date,
+    sample_id,
+    client_id,
+    event_timestamp,
+    normalized_channel,
+    normalized_country_code,
+    event_name,
+    search_session_type,
+    CASE
+      WHEN search_session_type = 'engaged'
+        THEN engagement_type
+      ELSE NULL
+    END AS engaged_result_type,
+    result_type,
+    number_of_chars_typed,
+    num_total_results,
+    CASE
+      WHEN search_session_type = 'annoyance'
+        THEN engagement_type
+      ELSE NULL
+    END AS annoyance_signal_type,
+    COUNTIF(
+      res IN ('search_suggest', 'search_history', 'search_suggest_rich')
+    ) AS num_default_partner_search_suggestions,
+    COUNTIF(res IN ('searchengine')) AS num_search_engine_suggestions_impressions,
```

Comment:
```suggestion
    COUNTIF(res IN ('search_engine')) AS num_search_engine_suggestions_impressions,
```

I think this was a typo in the speclet doc. The full list of possible types can be found in the [event doc](https://dictionary.telemetry.mozilla.org/apps/firefox_desktop/metrics/urlbar_engagement)

On a separate note, what about removing `suggestions` from the name, so changing this to `num_search_engine_impressions` @rebecca-burwei - this is what we had in the doc but I'm worried it'll be confusing.

Reply:
yeah, I think `num_search_engine_impressions` is too confusing. I feel strongly that Product should either choose these names, or be heavily bought in to them. Otherwise, they will just keep asking us what the names mean when it's really their responsibility to define the product experience. So, maybe you could go back to the doc and talk to Tito about it?

Reply:
I am going to close this thread and restart it in the UDF section.